backend/modules/metadata_store/truefoundry.py

```diff
@@ -529,3 +533,31 @@ def log_errors_for_data_ingestion_run(
             )
         except Exception as e:
             logger.exception(e)
+
+    async def list_collections(self) -> List[str]:
+        logger.info(f"[Metadata Store] Listing all collection")
+        ml_runs = self.client.search_runs(
+            ml_repo=self.ml_repo_name,
+            filter_string=f"params.entity_type = '{MLRunTypes.COLLECTION.value}'",
+        )
+        return [run.run_name for run in ml_runs]
+
+    async def list_data_sources(self) -> List[str]:
+        logger.info(f"[Metadata Store] Listing all data sources")
+        ml_runs = self.client.search_runs(
+            ml_repo=self.ml_repo_name,
+            filter_string=f"params.entity_type = '{MLRunTypes.DATA_SOURCE.value}'",
+        )
+
+        data_sources = []
+        for run in ml_runs:
+            run_params = run.get_params()
+            data_sources.append(
+                {
+                    "type": run_params.get("data_source_fqn").split("::")[0],
+                    "uri": run_params.get("data_source_fqn").split("::")[1],
+                    "fqn": run_params.get("data_source_fqn"),
+                    "metadata": run_params.get("metadata", {}),
```

Comment:
this metadata field will never be presend, so you can remove that

Reply:
done.